### PR TITLE
Use context wait patterns

### PR DESCRIPTION
**Pull request by jterry75**
Signed-off-by: Justin Terry (VM) <juterry@microsoft.com>

### CODE REVIEW

File: internal/storage/scsi/scsi.go [hunk 1]
```diff
@@ -96,22 +94,24 @@ func ControllerLunToName(ctx context.Context, controller, lun uint8) (_ string,
...
+	// under /sys/bus/scsi/devices/<scsiID>/block.
+	blockPath := filepath.Join("/sys/bus/scsi/devices", scsiID, "block")
 	var deviceNames []os.FileInfo
-	startTime := time.Now()
 	for {
-		// Devices matching the given SCSI code should each have a subdirectory
-		// under /sys/bus/scsi/devices/<scsiID>/block.
-		var err error
-		deviceNames, err = ioutil.ReadDir(filepath.Join("/sys/bus/scsi/devices", scsiID, "block"))
+		deviceNames, err = ioutil.ReadDir(blockPath)
```
Comment (kevpar) [inline internal/storage/scsi/scsi.go]: It looks like if this call succeeds we never exit the loop.

Comment (jterry75) [reply internal/storage/scsi/scsi.go]: I have no idea how this passed tests. Good catch. Let me fix and try again.

File: internal/runtime/hcsv2/uvm.go [hunk 2]
```diff
@@ -204,8 +205,10 @@ func newInvalidRequestTypeError(rt prot.ModifyRequestType) error {
 func modifyMappedVirtualDisk(ctx context.Context, rt prot.ModifyRequestType, mvd *prot.MappedVirtualDiskV2) (err error) {
 	switch rt {
 	case prot.MreqtAdd:
+		mountCtx, cancel := context.WithTimeout(ctx, time.Second*4)
```
Comment (kevpar) [inline internal/runtime/hcsv2/uvm.go]: FYI, looks like this timeout is changing from 2 to 4 seconds. Not sure if that's intentional.

Comment (jterry75) [reply internal/runtime/hcsv2/uvm.go]: There were two calls that each had 2 second timeouts. Now the entire set (both calls) have up to the total 4 seconds.

**Pull request merged**